Week2/HYF-JS3-HW2/index.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+{
+  function fetchJSON(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+      xhr.onload = () => {
+        if (xhr.status < 400) {
+          resolve(xhr.response);
+        } else {
+          reject(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        }
+      };
+      xhr.onerror = () => reject(new Error('Network request failed'));
+      xhr.send();
+    });
+  }
+
+  function createAndAppend(name, parent, options = {}) {
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    Object.keys(options).forEach(key => {
+      const value = options[key];
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  const root = document.getElementById('root');
+  const header = createAndAppend('header', root, { class: 'header' });
+  createAndAppend('p', header, { text: 'HYF Repositories' });
+
+  const select = createAndAppend('select', header, { id: 'selector' });
+  const option = x => {
```

Comment:
This is a function but I had to look hard to find it:

1. A function should always have a verb as its main word, to indicate the action that is performed by the function. In this case you are creating `<option>` elements, so this function might be named `createOptions`.
2. The parameter name `x` is meaningless. I had to look elsewhere in your code to find out that you are passing an array of repository objects to this function, therefore the parameter name should be `repositories`. (Reserve the name `x` for the indepedent in math.)

Reply:
I will correct it.